cli_helpers/tabular_output/delimited_output_adapter.py

```diff
@@ -27,10 +32,15 @@ def adapter(data, headers, table_format='csv', **kwargs):
     ckwargs = {'delimiter': delimiter}
     ckwargs.update(filter_dict_by_key(kwargs, keys))
 
-    with contextlib.closing(StringIO()) as content:
-        writer = csv.writer(content, **ckwargs)
-        writer.writerow(headers)
-        for row in data:
-            writer.writerow(row)
-
-        return content.getvalue()
+    l = linewriter()
+    writer = csv.writer(l, **ckwargs)
+    l.reset()
+    writer.writerow(headers)
+    for line in l.lines:
+        yield line[:-1]
```

Comment:
Also, `lineterminator` needs to be removed from line 23 since client code should not be able to specify a line ending (since we return a generator with no newlines).